### PR TITLE
Js assignment 11, 12

### PR DESCRIPTION
**Pull request by sanketmod**

### CODE REVIEW

File: sanket/javascript/assignment12(day25)/sanketA12_Q1.html [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+          document.write(
+            `employeeNo= ${this.empNo}, name= ${this.name}, address= ${this.address},
+            city= ${this.city}, salary= ${this.salary}<br>`
+          );
+        }
+        calculate() {
+          let da = 0.3 * this.salary;
+          let hra = 0.15 * this.salary;
+          let pf = 0.125 * this.salary;
+          this.gross = this.salary + da + hra;
```
Comment (MKadeer17) [inline sanket/javascript/assignment12(day25)/sanketA12_Q1.html:29]: remove this.salary and add basic salary like: this.basic = this.salary * 12
```suggestion
          this.basic = this.salary *12;
          this.gross = (this.basic) + da *12 + hra *12;
```

File: sanket/javascript/assignment12(day25)/sanketA12_Q1.html [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  </head>
+  <body>
+    <script>
+      class Employee {
+        constructor() {
+          this.empNo = 101;
+          this.name = "Ashok Kumar";
+          this.address = "43 Ramnagar";
+          this.city = "Pune Maharashtra";
+          this.salary = 80000;
```
Comment (MKadeer17) [inline sanket/javascript/assignment12(day25)/sanketA12_Q1.html:17]: **create da, hra, pf class variable**
for this.gross empty class variable
**like: this.da, etc.**

File: sanket/javascript/assignment12(day25)/sanketA12_Q2.html [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+  </head>
+  <body>
+    <script>
+      class Employee {
+        constructor(empNo, name, address, city, salary) {
+          this.empNo = empNo;
+          this.name = name;
+          this.address = address;
+          this.city = city;
+          this.salary = salary;
```
Comment (MKadeer17) [inline sanket/javascript/assignment12(day25)/sanketA12_Q2.html]: Same here also.

**Review by MKadeer17 — APPROVED** [review]

File: sanket/javascript/assignment12(day25)/sanketA12_Q1.html (hunk 2 above)
Comment (sanketmod) [reply sanket/javascript/assignment12(day25)/sanketA12_Q1.html:17]: if you are using variables like this.da is instance variable not class variable, also i am not going to use these variable outside, so creation of an instance variable is a very costly operation if you are not going to use it anywhere in the program do not create instance variable

File: sanket/javascript/assignment12(day25)/sanketA12_Q2.html (hunk 3 above)
Comment (sanketmod) [reply sanket/javascript/assignment12(day25)/sanketA12_Q2.html]: same here

File: sanket/javascript/assignment12(day25)/sanketA12_Q1.html (hunk 1 above)
Comment (sanketmod) [reply sanket/javascript/assignment12(day25)/sanketA12_Q1.html:29]: i think you are not getting the question, please read the question properly also, salary and basic salary, annual basic salary

Comment (MKadeer17) [reply sanket/javascript/assignment12(day25)/sanketA12_Q1.html:29]: Ok

File: sanket/javascript/assignment12(day25)/sanketA12_Q1.html (hunk 2 above)
Comment (MKadeer17) [reply sanket/javascript/assignment12(day25)/sanketA12_Q1.html:17]: Ok

File: sanket/javascript/assignment12(day25)/sanketA12_Q2.html (hunk 3 above)
Comment (MKadeer17) [reply sanket/javascript/assignment12(day25)/sanketA12_Q2.html:17]: Ok

**Review by MKadeer17 — APPROVED** [review]

**Review by poonam-khode — APPROVED** [review]

File: sanket/javascript/assignment11(day24)/sanketA11_Q1.html [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+          i++;
+          marks.push(m);
+        }
+      }
+      let total = [];
+      for (i = 0; i < marks.length; i++) {
+        t = parseInt(marks[i]) + parseInt((20 * marks[i]) / 100);
+        total.push(t);
+      }
+      let totalMarks = total.sort().reverse();
```
Comment (Snehal-113) [inline sanket/javascript/assignment11(day24)/sanketA11_Q1.html]: The RollNo has to be associated with the marks. If the user enters for example 35 for rollno 1 and after sorting 35 comes on 3rd position, then rollno 1 must also come on 3rd position.

File: sanket/javascript/assignment12(day25)/sanketA12_Q1.html (hunk 2 above)
Comment (Snehal-113) [reply sanket/javascript/assignment12(day25)/sanketA12_Q1.html:17]: yes, no need to create da,hra,pf as class properties

File: sanket/javascript/assignment12(day25)/sanketA12_Q1.html (hunk 1 above)
Comment (Snehal-113) [reply sanket/javascript/assignment12(day25)/sanketA12_Q1.html:29]: this.gross = this.salary + da + hra;
the previous statement is correct

File: sanket/javascript/assignment12(day25)/sanketA12_Q2.html (hunk 3 above)
Comment (Snehal-113) [reply sanket/javascript/assignment12(day25)/sanketA12_Q2.html]: need to create an empty constructor here, as per question

File: sanket/javascript/assignment12(day25)/sanketA12_Q2.html [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+            grossSalary= ${this.gross}, netSalary= ${this.net}<br><br>`
+          );
+        }
+      }
+      const empOne = new Employee(
+        this.empNo,
+        this.name,
+        this.address,
+        this.city,
+        this.salary
```
Comment (Snehal-113) [inline sanket/javascript/assignment12(day25)/sanketA12_Q2.html]: no parameters will be passed to the constructor.Properties are initialized in getData() method

**Review by Snehal-113 — CHANGES_REQUESTED** [review]

**Review by Snehal-113 — APPROVED** [review]

**Pull request merged**